### PR TITLE
Nakul/intellisense mysql

### PR DESCRIPTION
**Pull request by nakulms**

### CODE REVIEW

File: ossdbtoolsservice/language/completion/pg_completer.py [hunk 1]
```diff
@@ -12,20 +12,21 @@
...
-    Keyword, NamedQuery, Datatype, Alias, Path, JoinCondition, Join
-)
-from .packages.parseutils.meta import ColumnMetadata, ForeignKey
+
+from .packages.parseutils.pg_utils.meta import ColumnMetadata, ForeignKey
+from .packages.parseutils.pg_utils.tables import TableReference
 from .packages.parseutils.utils import last_word
-from .packages.parseutils.tables import TableReference
 from .packages.pgliterals.main import get_literals
-from .completer import MyCompleter, _Candidate, Match
```
Comment (rishky-msft) [inline ossdbtoolsservice/language/completion/pg_completer.py:28]: Can we check this file and add only required changes for PG. No additional refactoring required in this.

File: ossdbtoolsservice/language/metadata_executor.py [hunk 2]
```diff
@@ -90,3 +90,11 @@ def functions(self) -> List[tuple]:
         use the lightweight metadata query as it'll have N queries for N functions otherwise
         """
         return [f for f in self.lightweight_metadata.functions()]
+
+    def users(self) -> List[tuple]:
+        """return list of users, used for MySQLCompleter"""
```
Comment (rishky-msft) [inline ossdbtoolsservice/language/metadata_executor.py:95]: we can remove mysql completer reference from comment

File: ossdbtoolsservice/language/metadata_executor.py [hunk 3]
```diff
@@ -90,3 +90,11 @@ def functions(self) -> List[tuple]:
...
+        return [u for u in self.lightweight_metadata.users()]
+
+    def show_candidates(self) -> List[tuple]:
+        """return list of show command candidates, used for MySQLCompleter"""
```
Comment (rishky-msft) [inline ossdbtoolsservice/language/metadata_executor.py:99]: same as above

File: ossdbtoolsservice/language/completion_refresher.py [hunk 4]
```diff
@@ -34,7 +34,10 @@ class CompletionRefresher:
     completion suggestions in a background thread.
     """
 
-    refreshers = OrderedDict()
+    refreshers = {
+        PG_PROVIDER_NAME: OrderedDict(),
+        MYSQL_PROVIDER_NAME: OrderedDict()
```
Comment (rishky-msft) [inline ossdbtoolsservice/language/completion_refresher.py:39]: Can we use the same refresher for Mariadb and Mysql?
Suggestions?

File: ossdbtoolsservice/language/completion/mysql_completer.py [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+        :return:
+        """
+        # 'data' is a generator object. It can throw an exception while being
+        # consumed. This could happen if the user has launched the app without
+        # specifying a database name. This exception must be handled to prevent
+        # crashing.
+        try:
+            data = [self.escaped_names(d) for d in data]
+        except Exception:
+            data = []
```
Comment (rishky-msft) [inline ossdbtoolsservice/language/completion/mysql_completer.py:107]: Can we ensure that these exceptions are getting logged

Comment (rishky-msft) [reply ossdbtoolsservice/language/completion/mysql_completer.py:107]: same for all below

File: ossdbtoolsservice/language/completion/packages/parseutils/mysql_utils/mysql_utils.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    # abc is the table name, but if we don't stop at the first lparen, then
+    # we'll identify abc, col1 and col2 as table names.
+    insert_stmt = parsed[0].token_first().value.lower() == 'insert'
+    stream = extract_from_part(parsed[0], stop_at_punctuation=insert_stmt)
+    return list(extract_table_identifiers(stream))
+
+
+if __name__ == '__main__':
+    sql = 'select * from (select t. from tabl t'
+    print(extract_tables(sql))
```
Comment (rishky-msft) [inline ossdbtoolsservice/language/completion/packages/parseutils/mysql_utils/mysql_utils.py:102]: why are we printing this?

File: ossdbtoolsservice/language/metadata_executor.py (hunk 2 above)
Comment (nakulms) [reply ossdbtoolsservice/language/metadata_executor.py:95]: Done

File: ossdbtoolsservice/language/metadata_executor.py (hunk 3 above)
Comment (nakulms) [reply ossdbtoolsservice/language/metadata_executor.py:99]: Done

File: ossdbtoolsservice/language/completion/mysql_completer.py (hunk 5 above)
Comment (nakulms) [reply ossdbtoolsservice/language/completion/mysql_completer.py:107]: done

File: ossdbtoolsservice/language/completion/packages/parseutils/mysql_utils/mysql_utils.py (hunk 6 above)
Comment (nakulms) [reply ossdbtoolsservice/language/completion/packages/parseutils/mysql_utils/mysql_utils.py:102]: This is here to check the expected output on terminal by running mysql_utils.py directly. Can be ignored.

**Review by rishky-msft — APPROVED** [review]

File: ossdbtoolsservice/language/completion_refresher.py (hunk 4 above)
Comment (nakulms) [reply ossdbtoolsservice/language/completion_refresher.py:39]: Will have to investigate further for Mariadb. Making a note for it.

**Pull request merged**